### PR TITLE
Added Routes: #181

### LINKED ISSUE #181: Create routes with placeholders
This next iteration of the Atlas should have the following basic paths, where text in <angle_brackets> represents a variable:

| Name | Path |
| - | - |
| Home | / |
| Explore a dataset | /explore/<dataset_id> |
| View a single entry’s detail | /explore/<dataset_id>/<entry_id> |
| Compare entries | /explore/<dataset_id>/comparison |
| Use a guided tool | /tools/<tool_id> |
| Various static pages | /faq, /research, /contributors, etc. |

And there should be a blank placeholder page for each of those URLs, to make it easier to work on components in parallel later on. 

For now, capture but ignore the path variables in those placeholders. For example, the layout `/explore/<dataset_id>` will eventually vary depending on `<dataset_id>`, but for now let's treat `/explore/foo` and `/explore/bar` as the same page.

### PR DESCRIPTION
**Pull request by torosyaneric**
Added Routes: Contributors, Faq, Research, Dataset, Dataset Entry, Dataset Comparison, Guided Tool,

Updated package.json: added router deps,,
Updated .gitignore,

Resolves #181

### CODE REVIEW

File: frontend/src/routes/Contributors/Contributors.stories.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+import PATH from '../../utils/path'
+import Contributors from './Contributors'
+
+export default {
+  title: `routes/${PATH.CONTRIBUTORS}`,
```
Comment (justinanderson) [inline frontend/src/routes/Contributors/Contributors.stories.js:7]: Storybook fails to load if any *.stories.js file has a `title` ending with a slash or contain consecutive slashes. The console error for Contributors looks like `Uncaught (in promise) Error: Invalid part '', leading to id === parentId ('routes'), inside kind 'routes//contributors/'`.

You can fix this if you change this line to ``title: `routes${PATH.CONTRIBUTORS}`,`` and edit utils/paths.js to not have a trailing slash.

Same for all of the other route stories except Home.

File: frontend/src/routes/Home/Home.js [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+
         <a className="Home__link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
           Learn React
         </a>
+
+        <p>Temp router links</p>
+
+        <ul>
+          <li>
+            <Link to={PATH.DATASET.replace(':', '1')}>DATASET</Link>
```
Comment (justinanderson) [inline frontend/src/routes/Home/Home.js:24]: Storybook fails here, and will for any other page we add a Link to in the future, with the error `useHref() may be used only in the context of a <Router> component.`. That can be resolved by wrapping `<Home {...args} />` in a MemoryRouter.

File: frontend/src/routes/index.js [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+      <Route exact path={PATH.DATASET} element={<Dataset />} />
+
+      {/* tool */}
+      <Route exact path={PATH.GUIDED_TOOL} element={<GuidedTool />} />
+
+      {/* static */}
+      <Route exact path={PATH.FAQ} element={<Faq />} />
+      <Route exact path={PATH.RESEARCH} element={<Research />} />
+      <Route exact path={PATH.CONTRIBUTORS} element={<Contributors />} />
+    </Switch>
```
Comment (justinanderson) [inline frontend/src/routes/index.js:36]: Let's add a 404 default route as well, so it's obvious when a URL hasn't matched any routes.

**Review by justinanderson — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Contributors/Contributors.stories.js (hunk 1 above)
Comment (torosyaneric) [reply frontend/src/routes/Contributors/Contributors.stories.js]: Fixed

File: frontend/src/routes/Home/Home.js (hunk 2 above)
Comment (torosyaneric) [reply frontend/src/routes/Home/Home.js:24]: Fixed using a decorator

File: frontend/src/routes/index.js (hunk 3 above)
Comment (torosyaneric) [reply frontend/src/routes/index.js:40]: Done

File: frontend/src/routes/index.js [hunk 4]
```diff
@@ -33,6 +34,9 @@ function Routes() {
       <Route exact path={PATH.FAQ} element={<Faq />} />
       <Route exact path={PATH.RESEARCH} element={<Research />} />
       <Route exact path={PATH.CONTRIBUTORS} element={<Contributors />} />
+
+      <Route exact path={PATH[404]} element={<NotFound />} />
+      <Route path="*" element={<Navigate to={PATH[404]} />} />
```
Comment (justinanderson) [inline frontend/src/routes/index.js:39]: Oh sorry, I didn't mean to redirect to a 404 page. Just to show a 404 page for any unmatched path. I find it useful during development to have the location remain when it's invalid, so I can tell what's wrong about it.

**Review by justinanderson — CHANGES_REQUESTED** [review]

File: frontend/src/routes/index.js (hunk 4 above)
Comment (torosyaneric) [reply frontend/src/routes/index.js]: Got it. Fixed

File: frontend/src/routes/index.js [hunk 5]
```diff
@@ -1,2 +1,44 @@
-export { default } from './Home'
-// TODO: Please add react-router-dom here.
+// vendors
+import React from 'react'
+import { Navigate, Route, Routes as Switch } from 'react-router-dom'
```
Comment (justinanderson) [inline frontend/src/routes/index.js:3]: Unused import of Navigate

**Review by justinanderson — CHANGES_REQUESTED** [review]

**Review by justinanderson — APPROVED** [review]

**Pull request merged**